webui/react/src/pages/ExperimentDetails.tsx

```diff
@@ -49,34 +47,38 @@ const ExperimentDetails: React.FC = () => {
   const [ tabKey, setTabKey ] = useState(defaultTabKey);
   const [ forkModalVisible, setForkModalVisible ] = useState(false);
   const [ forkModalConfig, setForkModalConfig ] = useState('Loading');
-  const [ source ] = useState(axios.CancelToken.source());
-  const [ experimentDetails, setExperimentDetails ] = useState<ApiState<ExperimentBase>>({
-    data: undefined,
-    error: undefined,
-    isLoading: true,
-    source,
-  });
-  const [ experimentCanceler ] = useState(new AbortController());
+  const [ canceler ] = useState(new AbortController());
+  const [ experiment, setExperiment ] = useState<ExperimentBase>();
   const [ valHistory, setValHistory ] = useState<ValidationHistory[]>([]);
+  const [ pageError, setPageError ] = useState<Error>();
 
   const id = parseInt(experimentId);
   const basePath = paths.experimentDetails(experimentId);
-  const experiment = experimentDetails.data;
 
   const fetchExperimentDetails = useCallback(async () => {
     try {
-      const experiment = await getExperimentDetails({ id }, { signal: experimentCanceler.signal });
-      const validationHistory = await getExpValidationHistory({ id });
-      if (!isEqual(experiment, experimentDetails.data)) {
-        setExperimentDetails(prev => ({ ...prev, data: experiment, isLoading: false }));
+      const [ experimentData, validationHistory ] = await Promise.all([
+        getExperimentDetails({ id }, { signal: canceler.signal }),
+        getExpValidationHistory({ id }, { signal: canceler.signal }),
+      ]);
+      if (!isEqual(experimentData, experiment)) {
```

Comment:
I guess you already know that, `isEqual` JSON comparison might fail in many ways, but is acceptable for those cases? (and worst case scenario will trigger an un-needed re-render?)

Reply:
Yeah we should probably update this function. I'll add a ticket for this.